core/src/main/java/org/opensearch/sql/ast/expression/DataType.java

```diff
@@ -26,7 +26,9 @@ public enum DataType {
   DATE(ExprCoreType.DATE),
   TIME(ExprCoreType.TIME),
   TIMESTAMP(ExprCoreType.TIMESTAMP),
-  INTERVAL(ExprCoreType.INTERVAL);
+  INTERVAL(ExprCoreType.INTERVAL),
+
+  IP(ExprCoreType.IP);
```

Comment:
Cast_to_IP unsupported yet since IP type is not supported with Calcite. But this is a missing code in DataType class.